lib/modules.nix

```diff
@@ -1256,7 +1256,68 @@ let
       (opt.highestPrio or defaultOverridePriority)
       (f opt.value);
 
-  doRename = { from, to, visible, warn, use, withPriority ? true, condition ? true }:
+  /*
+    Return a module that help declares an option that has been renamed.
+    When a value is defined for the old option, it is forwarded to the `to` option.
+   */
+  doRename = {
+    # List of strings representing the attribute path of the old option.
+    from,
+    # List of strings representing the attribute path of the new option.
+    to,
+    # Boolean, whether the old option is to be included in documentation.
+    visible,
+    # Whether to warn when a value is defined for the old option.
+    # NOTE: This requires the NixOS assertions module to be imported, so
+    #        - this generally does not work in submodules
+    #        - this may or may not work outside NixOS
+    warn,
+    # A function that is applied to the merged option value, to form the value
+    # of the old `from` option. Example: ``lib.warn "Obsolete option `${opt.old}' is used. Use `${opt.to}' instead."``
```

Comment:
What does "merged" mean here? Reading only the docs for this function I couldn't quite understand this part or why it's needed. If in most cases people just want an identity function here (like in the example below), it might be useful to also mention that.

Reply:
https://github.com/NixOS/nixpkgs/compare/4833d64733a846ec0e89209af386878e62f3ad3a..fe36252019a7b21c951825f7028d4ef7b46c99a8